app/models/dossier.rb

```diff
@@ -235,10 +235,10 @@ def classer_sans_suite(motivation: nil, instructeur: nil, processed_at: Time.zon
   scope :en_instruction,              -> { not_archived.state_en_instruction }
   scope :termine,                     -> { not_archived.state_termine }
 
-  scope :processed_in_month, -> (month) do
+  scope :processed_in_month, -> (date) do
+    date = date.to_datetime # ensure casting of date to datetime, otherwise when given a date, compared in postgres it is casted as datetime at midnight
     state_termine
-      .joins(:traitements)
-      .where(traitements: { processed_at: month.beginning_of_month..month.end_of_month })
+      .where("processed_at <= ?", date.end_of_month)
```

Comment:
ah j ai peur que processed_at soit un relicat qu'il faille supprimer, l'info est présente sur les traitements

Reply:
merci pour ton retour @LeSim. mon dernier commit etait vraiment nul. 

pour répondre a ton point ; j'ai l'impression que les `traitements` permettent de gérer l'historique pour le coté "audit log" de ce qui se passe sur un dossier. (@tchak m'avait aguillé sur un refacto qui aille dans ce sens lorsque j'ai bossé sur le sujet de l'expiration des dossiers). bref, c'est un nice to have que de virer cette jointure sur cette PR. Cependant je pense qu'il faudrait le faire sur le reste de l'app. 

pour en revenir au fix en soit ; grace a ton retour je me rends compte que c'est du code très défensif : j'aime pas. je reviens avec une meilleur proposition :-)

Reply:
Tu as raison d'utiliser `dossiers.processed_at`. À un moment nous avons supprimé `dossiers.processed_at` au profit de `traitements.processed_at` mais nous sommes revenues en arrière. `dossiers.processed_at` correspond à la date du dernier traitement. C'est ce qu'il faut utilisé dans la plupart des cas.